nixos/modules/services/x11/xserver.nix

```diff
@@ -219,6 +219,12 @@ in
         '';
       };
 
+      dpi = mkOption {
+        type = types.nullOr types.int;
+        default = 0;
```

Comment:
You probably want to default to `null` instead, because later on you're checking explicitly for `null` and not `0`.